### ISSUE TITLE
Precision drop when using --input-as-nchw

### ISSUE THREAD

**Issue by romainver** [onnx/tensorflow-onnx#993]
Hello,

I am trying to convert multiple TensorFlow classification models into ONNX format using tf2onnx.
Those models have been trained on a binary classification problem and performs well (+90% on test set).
They have been trained with channels as last (NHWC).

Those models are stored on my disk in the TensorFlow SavedModel format.
When I convert those models into ONNX using tf2onnx.convert, it works fine.
I then measure accuracy of the generated ONNX models using ONNXRuntime and TensorRT.
Results on test set stay the same (around 90%).

When I convert with the --inputs-as-nchw argument with my input node name, it works fine.

I then can measure the generated ONNX model accuracy with ONNXRuntime or TensorRT.
On the same test set used before the precision drop around 50%.

Problem is the same for VGG16, Resnet152, and xception.
I preprocessed my data into channel first using np.reshape(N,3,H,W), I don't know if that can be an issue.

I found this related [issue](https://github.com/Unity-Technologies/barracuda-release/issues/64#issuecomment-647015670).

Let me know if you need any models I have tried with a sample of my dataset. 

Thanks in advance for your help

**Comment by jignparm** [comment]
`np.reshape()` is probably not sufficient to convert between channels-last format and channels-first format. Try  `data = np.transpose(data, axes=[0,3,1,2])` instead. 

This should solve the 50% drop in precision/accuracy. Please update on the results.

**Comment by romainver** [comment]
Thanks a lot it works now.
